### ISSUE TITLE
[3.6.0] Extend 'Initial Layer Horizontal Expansion' over a specified number of layers instead of only 1

### ISSUE THREAD

**Issue by mcd8604** [Ultimaker/CuraEngine#994]
Often, the "elephant's foot" effect is not mitigated sufficiently by setting the 'Initial Layer Horizontal Expansion' to a negative value. This value is effectively limited by the nozzle diameter and is unsafe to exceed beyond a typical safe overhang angle. To mitigate larger cases of elephant's foot, it would make sense to allow for larger values of negative polygon offset by spreading the expansion across a specified number of layers. 

For example, given the following settings, the second layer would print the outer wall on air:
- Nozzle Width = 0.4
- xy_offset_layer_0 = -0.6

If however we specify say, num_xy_offset_layers=3, an expansion could be performed as such:

(in expanded loop notation)
- if (layer_nr == 0) then xy_offset = -0.6
- if (layer_nr == 1) then xy_offset = -0.4
- if (layer_nr == 2) then xy_offset = -0.2

(generalized)
- offset_per_layer = xy_offset_layer_0 / num_xy_offset_layers
- xy_offset = (num_xy_offset_layers - layer_nr) * offset_per_layer 

Does this feature make sense? 
Are there better ways of compensating for elephant's foot?
Is there a better way to implement this?
Would these changes break any other features?

Thanks,

Mike

**Comment by Ghostkeeper** [comment]
Good idea. Fan speed has a similar type of "initial layer" fan speed that gradually goes to the ordinary fan speed over the course of several layers.

But we don't think that the effect is big enough to warrant priority over other things. We'll defer it, so maybe we'll come to this or a similar solution eventually. But not any time soon.
